src/engine/SumOfInfeasibilitiesManager.cpp

```diff
@@ -136,6 +143,23 @@ void SumOfInfeasibilitiesManager::initializePhasePatternWithCurrentInputAssignme
     }
 }
 
+void SumOfInfeasibilitiesManager::initializePhasePatternWithCurrentAssignment()
+{
+    obtainCurrentAssignment();
+
+    for ( const auto &plConstraint : _plConstraints )
+    {
+        ASSERT( !_currentPhasePattern.exists( plConstraint ) );
+        if ( plConstraint->isActive() && !plConstraint->phaseFixed() &&
+             plConstraint->getType() != DISJUNCTION )
```

Comment:
I don't like the SoIManager probing for exact constraint types. Better have something like a virtual function plc->supportsSoi(), and have the SoIManager call it to check